### PR TITLE
EVG-18231 Include author_email in patch created versions

### PR DESCRIPTION
**Pull request by khelif96**
[EVG-18231](https://jira.mongodb.org/browse/EVG-18231)

### Description 
author_email was not previously included in versions that were created from a patch. This now includes the email from their git account by default if not it falls back to the email included in gitinfo. 

### Testing 
Example doc of a patch I created on staging
image

### CODE REVIEW

**Review by malikchaya2 — CHANGES_REQUESTED** [review]
The code looks good, but I think we should test this on staging before merging

File: model/patch_lifecycle.go [hunk 1]
```diff
@@ -452,6 +452,17 @@ func FinalizePatch(ctx context.Context, p *patch.Patch, requester string, github
 		return nil, errors.Wrap(err, "fetching patch parameters")
 	}
 
+	authorEmail := p.GitInfo.Email
+	if p.GithubPatchData.AuthorUID != 0 {
+		u, err := user.FindByGithubUID(p.GithubPatchData.AuthorUID)
+		grip.Error(message.WrapError(err, message.Fields{
```
Comment (ablack12) [inline model/patch_lifecycle.go:458]: I'd rather not log this if we aren't going to do anything with it; it's not really an error i think that a user might not have included this (although I see we use this same format elsewhere; we may just want to remove them both, or at least change them to info)

Comment (khelif96) [reply model/patch_lifecycle.go:458]: Done just removed the error here.

**Comment by khelif96** [comment]
> The code looks good, but I think we should test this on staging before merging

Done just tested on staging and attached the document that resulted.

File: model/patch_lifecycle.go [hunk 2]
```diff
@@ -452,6 +452,14 @@ func FinalizePatch(ctx context.Context, p *patch.Patch, requester string, github
 		return nil, errors.Wrap(err, "fetching patch parameters")
 	}
 
+	authorEmail := p.GitInfo.Email
```
Comment (ybrill) [inline model/patch_lifecycle.go:455]: The patch's Author field and the corresponding version's Author/AuthorID correspond to the id of the user who created the patch. 
Could it work to fetch the user with that instead?

**Review by ybrill — COMMENTED** [review]
feel free to ignore me 😅 
I was already thinking about this so I figured I'd look at the PR

**Review by malikchaya2 — APPROVED** [review]

File: model/patch_lifecycle.go (hunk 2 above)
Comment (ablack12) [reply model/patch_lifecycle.go:455]: That's a good idea since it may work for more kinds of patches

Comment (khelif96) [reply model/patch_lifecycle.go:455]: Good point sure!

File: model/patch_lifecycle.go [hunk 3]
```diff
@@ -452,6 +452,14 @@ func FinalizePatch(ctx context.Context, p *patch.Patch, requester string, github
...
+	if p.Author != "" {
+		u, _ := user.FindOneById(p.Author)
```
[the remark this answers is not in the document]
Comment (khelif96) [reply model/patch_lifecycle.go]: Sure added!

**Review by ybrill — APPROVED** [review]
lgtm too! 😄

**Pull request merged**